lib/Optimizer/CodeGen/ConvertToQIR.cpp

```diff
@@ -2026,6 +2033,9 @@ void cudaq::opt::initializeTypeConversions(LLVMTypeConverter &typeConverter) {
     if (isa<NoneType>(eleTy))
       return factory::getPointerType(type.getContext());
     eleTy = typeConverter.convertType(eleTy);
+    if (isa<NoneType>(eleTy))
+      return factory::getPointerType(type.getContext());
+
```

Comment:
Why doesn't the recursion handle this? The recursive call on line 2035 ought to handle pointers to pointers.